### PR TITLE
Add Causely to adopters.yaml

### PR DESCRIPTION
**Pull request by esara**
Read  [LinkedIn post](https://www.linkedin.com/posts/jpkroehling_opentelemetry-graduation-by-austinlparker-activity-7172873030925332480-FvMW) from @jpkrohling and requested to include us in the adopters list,
as we have been using, promoting and contributing to OpenTelemetry.

Thanks!

### CODE REVIEW

File: data/ecosystem/vendors.yaml [hunk 1]
```diff
@@ -41,6 +41,13 @@
   contact: 'hello@betterstack.com'
   oss: false
   commercial: true
+- name: Causely
+  distribution: false
+  nativeOTLP: true
+  url: 'https://www.causely.io/blog/unveiling-the-causal-revolution-in-observability/'
```
Comment (svrnm) [inline data/ecosystem/vendors.yaml:47]: please point to your documentation around OTLP

File: data/ecosystem/vendors.yaml [hunk 2]
```diff
@@ -41,6 +41,13 @@
...
+  contact: 'early-access@causely.io'
```
Comment (svrnm) [inline data/ecosystem/vendors.yaml:48]: this contact is for us to follow up with you if we see any issues like a broken link, a github handle is preferred, an email address is OK as well.

File: data/ecosystem/adopters.yaml
Comment (svrnm) [inline data/ecosystem/adopters.yaml:1]: Please put this in a separate PR, we need to discuss if/how we handle vendor's dog fooding opentelemetry.

**Review by svrnm — CHANGES_REQUESTED** [review]

File: data/ecosystem/vendors.yaml (hunk 2 above)
Comment (esara) [reply data/ecosystem/vendors.yaml]: changed it to our support email

File: data/ecosystem/vendors.yaml (hunk 1 above)
Comment (esara) [reply data/ecosystem/vendors.yaml]: done

File: data/ecosystem/adopters.yaml
Comment (esara) [reply data/ecosystem/adopters.yaml:1]: sure, I can break this into two prs

Comment (esara) [reply data/ecosystem/adopters.yaml:1]: done

**Review by jpkrohling — CHANGES_REQUESTED** [review]
Thank you for responding to my call! It's great to hear you are not only a vendor, but also an end-user. 

We are trying to keep this page containing only organizations that are purely end-users, and therefore, I'd say that it's appropriate to have Causely listed only on the vendors page. I approved the other PR already.

**Comment by theletterf** [comment]
Closing as #4158 continues this.
